avm/res/azure-stack-hci/logical-network/main.bicep

```diff
@@ -0,0 +1,195 @@
+metadata name = 'Azure Stack HCI Logical Network'
+metadata description = 'This module deploys an Azure Stack HCI Logical Network.'
+
+// ============== //
+//   Parameters   //
+// ============== //
+
+@description('Required. Name of the resource to create.')
+param name string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Required. The custom location ID.')
+param customLocationId string
+
+@description('Required. The VM switch name.')
+param vmSwitchName string
+
+@description('Optional. The subnet name.')
+param subnet0Name string = 'default'
+
+@description('Optional. The IP allocation method.')
+@allowed([
+  'Static'
+  'Dynamic'
+])
+param ipAllocationMethod string = 'Dynamic'
+
+@description('Optional. The DNS servers list.')
+param dnsServers array = []
+
+@description('Optional. Tags for the logical network.')
+param tags object?
+
+@description('Optional. Address prefix for the logical network.')
+param addressPrefix string?
+
+@description('Optional. VLan Id for the logical network.')
+param vlanId int?
+
+@description('Optional. A list of IP configuration references.')
+param ipConfigurationReferences array = [
+  /*array of type {ID: string}*/
+]
```

Comment:
Instead of the comment, it would by highly recommnded to instead use a user-defined type like

```bicep
param ipConfigurationReferences ipConfigurationReferenceType[]?

@export()
@description('The type for an IP configuration reference.')
type ipConfigurationReferenceType = {
  @description('Required. The resource ID of something')
  resourceId: string
}

resource logicalNetwork 'Microsoft.AzureStackHCI/logicalNetworks@2024-05-01-preview' = {
  (...)
          ipConfigurationReferences: map((ipConfigurationReferences ?? []), (ipConfigurationReference) => { id: ipConfigurationReference.resourceId })
  (...)
}
```